prov/verbs/src/verbs_cm.c

```diff
@@ -123,23 +123,42 @@ vrb_msg_ep_prepare_cm_data(const void *param, size_t param_size,
 
 static inline void
 vrb_ep_prepare_rdma_cm_param(struct rdma_conn_param *conn_param,
-				struct vrb_cm_data_hdr *cm_hdr,
-				size_t cm_hdr_data_size)
+				void *priv_data, size_t priv_data_size)
 {
-	conn_param->private_data = cm_hdr;
-	conn_param->private_data_len = (uint8_t)cm_hdr_data_size;
+	conn_param->private_data = priv_data;
+	conn_param->private_data_len = (uint8_t)priv_data_size;
 	conn_param->responder_resources = RDMA_MAX_RESP_RES;
 	conn_param->initiator_depth = RDMA_MAX_INIT_DEPTH;
 	conn_param->flow_control = 1;
 	conn_param->rnr_retry_count = 7;
 }
 
+static void
+vrb_msg_ep_prepare_rdma_cm_hdr(void *priv_data,
+				const struct rdma_cm_id *id)
+{
+	struct vrb_rdma_cm_hdr *rdma_cm_hdr = priv_data;
+
+	rdma_cm_hdr->ip_version = 6 << 4; /* IPv6 */
+	rdma_cm_hdr->port =
+		(uint16_t)((struct ofi_sockaddr_ib *)&id->route.addr.src_addr)->sib_sid;
```

Comment:
Should call ofi_addr_get_port() for this

Reply:
I initially wanted to use ofi_addr_get_port() when I realized that the function returns the port in host byte order. 
Shouldn't we set the port in network byte order?

Reply:
I agree, it should be in network order.  But does this cast actually work?  The SID is in network byte order.

Reply:
Hm... Let me verify if the cast works.

Reply:
Ok, you are right and my cast actually doesn't work :)
I have updated the code to set the port with` htons(ofi_addr_get_port())`